srcpkgs/givaro/template

```diff
@@ -0,0 +1,38 @@
+# Template file for 'givaro'
+pkgname=givaro
+version=4.1.1
+revision=1
+build_style="gnu-configure"
+makedepends="gmpxx-devel"
+short_desc="C++ library for arithmetic and algebraic computations"
+maintainer="Gonzalo Tornaría <tornaria@cmat.edu.uy>"
+license="CECILL-B"
+homepage="https://github.com/linbox-team/givaro"
+distfiles="https://github.com/linbox-team/givaro/releases/download/v${version}/givaro-${version}.tar.gz"
+checksum=628049899386e91da245aee6cd446350fbca87e94863bc0d815066c08150487f
+nocross=yes
+
+build_options="native"
+
+if [ ! "$build_option_native" ]; then
+	configure_args="--enable-sse --enable-sse2
+	  --disable-sse3 --disable-ssse3 --disable-sse41 --disable-sse42
```

Comment:
I would rather remove option and make in unconditional. People are free to use XBPS_CFLAGS="-march=native" to get optimized build. Works for this package, should work for others too.

Reply:
I think the correct variable is `XBPS_TARGET_CFLAGS`? At least that's what's used in the `build-profiles` (the README there needs to be fixed, btw).

I agree that that's a better way of solving it, but it would be nice to make sure it's working. Also see #29732

Reply:
I mean XBPS_CFLAGS in etc/conf

Reply:
It might work in this case (assuming the only effect of the --enable-sse* configure flags is to add -msse* to CFLAGS, I'd have to check).

However, some math packages have custom written asm for different architectures so selecting build option "native" will not necessarily be equivalent to just using `-march=native`.

Another point is that using a build option will make an annotation in the xbps, e.g.:
```
$ xq givaro | grep native
build-options: ~native 
```
vs.
```
$ xbps-query -S givaro | grep build
build-options: native 
```